src/python/pants/backend/go/goals/debug_goals.py

```diff
@@ -114,6 +121,74 @@ async def dump_go_import_paths_for_module(
     return DumpGoImportPathsForModule(exit_code=0)
 
 
+class GoExportCgoCodegenSubsystem(GoalSubsystem):
+    name = "go-export-cgo-codegen"
```

Comment:
I think this makes sense to be a dedicated goal, only activated via the `debug` backend, rather than integrating into `export-codegen` or `export`. I agree with you that users usually will not want Cgo to be generated with those latter goals.